lib/db/mem.js

```diff
@@ -1042,6 +1055,28 @@ module.exports = function (log, error) {
     }
   }
 
+  Memory.prototype.accountRecord = function (emailBuffer) {
+    const normalizedEmail = emailBuffer.toString('utf8').toLowerCase()
+
+    if (! emails[normalizedEmail]) {
+      return P.reject(error.notFound())
+    }
+
+    const uid = emails[normalizedEmail].uid
+    return P.all([this.accountEmails(uid), this.account(uid)])
+      .spread((emails, account) => {
+
+        Object.keys(emails).some((key) => {
+          var emailRecord = emails[key]
+          if (emailRecord.uid.toString('hex') === uid.toString('hex') && emailRecord.isPrimary) {
+            account.primaryEmail = emailRecord.normalizedEmail
```

Comment:
Does this modify the stored `account` object in-place? If so that seems like a side-effect which isn't present in the equivalent stored procedure, can it lead to any observable differences in behaviour?